ratik/src/main/java/org/opencadc/ratik/Main.java

```diff
@@ -0,0 +1,202 @@
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.ratik;
+
+import ca.nrc.cadc.db.ConnectionConfig;
+import ca.nrc.cadc.db.DBUtil;
+import ca.nrc.cadc.util.Log4jInit;
+import ca.nrc.cadc.util.MultiValuedProperties;
+import ca.nrc.cadc.util.PropertiesReader;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import javax.naming.NamingException;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.InventoryUtil;
+import org.opencadc.inventory.db.SQLGenerator;
+import org.opencadc.inventory.util.ArtifactSelector;
+import org.opencadc.inventory.util.BucketSelector;
+
+/**
+ * Main entry point for ratik.
+ *
+ * @author pdowler
+ */
+public class Main {
+
+    private static final Logger log = Logger.getLogger(Main.class);
+    private static final String CONFIG_FILE_NAME = "ratik.properties";
+    private static final String CONFIG_PREFIX = Main.class.getPackage().getName();
+    private static final String SQLGENERATOR_CONFIG_KEY = SQLGenerator.class.getName();
+    private static final String DB_SCHEMA_CONFIG_KEY = CONFIG_PREFIX + ".db.schema";
+    private static final String LOGGING_KEY = CONFIG_PREFIX + ".logging";
+    private static final String DB_URL_CONFIG_KEY = CONFIG_PREFIX + ".db.url";
+    private static final String DB_USERNAME_CONFIG_KEY = CONFIG_PREFIX + ".db.username";
+    private static final String DB_PASSWORD_CONFIG_KEY = CONFIG_PREFIX + ".db.password";
+    private static final String QUERY_SERVICE_CONFIG_KEY = CONFIG_PREFIX + ".queryService";
+    private static final String URI_BUCKETS_CONFIG_KEY = CONFIG_PREFIX + ".buckets";
+    private static final String TRACK_SITE_LOCATIONS_CONFIG_KEY = CONFIG_PREFIX + ".trackSiteLocations";
+    private static final String ARTIFACT_SELECTOR_CONFIG_KEY = CONFIG_PREFIX + ".artifactSelector";
```

Comment:
For defined interface and plugin implementation, the key is the fully qualified class name of the interface and should be specified as code:
```
ArtifactSelector.class.getName()
```